windows/hid.c

```diff
@@ -409,6 +409,112 @@ static void* hid_internal_get_device_interface_property(const wchar_t* interface
 	return property_value;
 }
 
+static int hid_internal_extract_int_token_value(wchar_t* string, const wchar_t* token)
+{
+	int token_value;
+	wchar_t* startptr, * endptr;
+
+	/* Normalize to upper case */
+	for (wchar_t* p = string; *p; ++p) *p = towupper(*p);
```

Comment:
use similiar code instead of towupper/wcsstr calls?
```cpp
   size_t i;
    unsigned char c0 = *pattern, c1, c2;

    if (c0 == '\0')
        return (char *)str;

    c0 = toupper(c0);
    for (; (c1 = *str) != '\0'; str++) {
        if (toupper(c1) == c0) {
            for (i = 1;; i++) {
                c2 = pattern[i];
                if (c2 != '\0')
                    return (char *)str;
                c1 = str[i];
                if (toupper(c1) != toupper(c2))
                    break;
            }
        }
    }
```
 https://stackoverflow.com/a/35674861/1795050

Reply:
sure
want to do it within this PR or later?

Reply:
I'll do it myself.